### PR TITLE
resolved issue 4032 and added a unit test

### PR DESCRIPTION
**Pull request by wsw2016**

### CODE REVIEW

**Comment by wsw2016** [comment]
I signed it!

**Comment by acozzette** [comment]
ok to test (this comment will start the other CI runs)

File: src/google/protobuf/util/json_util.cc [hunk 1]
```diff
@@ -50,7 +50,8 @@ namespace util {
 
 namespace internal {
 ZeroCopyStreamByteSink::~ZeroCopyStreamByteSink() {
-  stream_->BackUp(buffer_size_);
+  if (buffer_size_ > 0)
```
Comment (acozzette) [inline src/google/protobuf/util/json_util.cc]: Could you put in curly braces around the body of the if statement, i.e.

if (buffer_size_ > 0 ) {
  ...
}

We usually go with that convention just to be safe.

File: src/google/protobuf/util/json_util_test.cc [hunk 2]
```diff
@@ -457,6 +458,20 @@ TEST(ZeroCopyStreamByteSinkTest, TestAllInputOutputPatterns) {
   }
 }
 
+TEST_F(JsonUtilTest, TestWronJsonInput) {
+  const char json[] = "{\"unknown_field\":\"some_value\"}";
```
Comment (acozzette) [inline src/google/protobuf/util/json_util_test.cc]: Couple small things:
- Add line breaks to keep the lines under 80 characters long
- Our convention is to use `snake_case` for variables in C++ so `protoBuffer` should be `proto_buffer` and likewise for `messageType`, and `resultStatus`.

**Pull request merged**